types/module/configurator.go

```diff
@@ -15,16 +19,31 @@ type Configurator interface {
 	// QueryServer returns a grpc.Server instance which allows registering services
 	// that will be exposed as gRPC services as well as ABCI query handlers.
 	QueryServer() grpc.Server
+
+	// RegisterMigration registers an in-place store migration for a module. The
+	// handler is a migration script to perform in-place migrations from version
+	// `fromVersion` to version `fromVersion+1`.
+	RegisterMigration(moduleName string, fromVersion uint64, handler func(store sdk.KVStore) error) error
```

Comment:
I'm thinking if we need `codec.BinaryMarshaler` here too (I don't think so for 0.42, but probably for future migrations)

Reply:
It's not needed. We can use `proto.Marshal`.

Reply:
So `proto.Marshal` only won't be enough, because we need the interfaceRegistry for decoding.

The question is more: do we need the whole proto binary encoder/decoder for migrations.

Reply:
Modules should have `codec.Marshaler` by this point, but in the regen ADR 033 implementation `Configurator` has `Marshaler` field for this purpose. So I would say let's add `Marshaler` to `Configurator`.

Reply:
And yes `proto.Marshal` is not enough

Reply:
OK, I added Marshaler to Configurator. Is `BinaryMarshaler` enough though?